### ISSUE TITLE
Alternate way of starting koreader on reMarkable?

### ISSUE THREAD

**Issue by hohusum** [koreader/koreader#6042]
Would it be possible to provide an alternate way of starting koreader besides a long press on the middle key on the reMarkable? This interferes with the use of the [binary patches to Xochitl](https://github.com/ddvk/remarkable-hacks) which rely on the middle button for fast switching between documents and notebooks. Also, the reMarkable 2, due this summer, will not have physical keys.

**Comment by pazos** [comment]
Use https://github.com/dixonary/draft-reMarkable. If you manage to make it working (it should be easy), please help us update wiki instructions in https://github.com/koreader/koreader/wiki/Installation-on-ReMarkable

**Comment by pazos** [comment]
Also using ssh directly, calling `systemctl start koreader` will kill the stock FW, start KO and start stock again after KO is down.

**Comment by hohusum** [comment]
But none of these alternatives would deactivate the binding to long press on the middle key?

**Comment by mergen3107** [comment]
IMO, once you provide another method of starting KOReader working, then there will be a good reason to get rid of the (only AFAIK) existing method.

**Comment by pazos** [comment]
> But none of these alternatives would deactivate the binding to long press on the middle key?

`systemctl disable button-listen`

**Comment by pazos** [comment]
@hohusum : please keep us updated. Does the launcher works? Could you update the wiki?. It would be cool to document the steps you took.

**Comment by hohusum** [comment]
This recipe for installing the draft launcher in conjunction with the plato
ereader should be a promising starting point:

https://www.reddit.com/r/RemarkableTablet/comments/bkbps9/how_to_install_plato_reader_and_add_it_to_draft/?st=jvp0mqus&sh=0651e353

Unfortunately the very starting point of the process requires building the
draft launcher on linux. I am not a linux user. Switching to another OS to
build a launcher to start an application makes the entry price too high for
me.

--hoh

**Comment by pazos** [comment]
Errrr, a binary is available [here](https://github.com/dixonary/draft-reMarkable/issues/3#issuecomment-492601607).

**Comment by hohusum** [comment]
But that was built May 5 2019, using the previous version of the reMarkable
toolchain. The current version of the reMarkable toolchain (link below) is
dated September 23 2019. Is there not a risk for complications if you drop
this binary into the most recent version?

https://remarkable.engineering/oecore-x86_64-cortexa9hf-neon-toolchain-zero-gravitas-1.8-23.9.2019.sh

**Comment by hohusum** [comment]
Sorry, when I look inside the zip archive I see it was built January 20
2019.

**Comment by pazos** [comment]
I don't know, I don't have the device. No contributors have the device.

@tcrs contributed the port (and I think he's fine with current launcher as it was his contribution too). @dfhall fixed wacom input but didn't raise concers about current launcher.

Launcher is normally an independent program outside KOReader. You might want to keep in touch with draft's makers, specially https://github.com/dixonary/draft-reMarkable/issues/6 or request the hability to launch 3rd party apps to the makers of binary patches.

TL;DR: not an internal KOReader issue. Keep issue open as documentation but should be "fixed" elsewhere.

**Comment by NiLuJe** [comment]
It's unlikely to be an issue. If it were built with a newer sdk than your firmware, maybe, but the other way around is usually safe.

**Comment by tcrs** [comment]
Whilst I don't think this resolves the issue I have added some more documentation for anyone who wants to embark on integrating with a custom launcher or creating their own: https://github.com/koreader/koreader/pull/6397

**Comment by hohusum** [comment]
I have added  installation instructions to the wiki.
